tests/ui/consts/do-not-ice-on-field-access-of-err-type.rs

```diff
@@ -0,0 +1,27 @@
+use std::mem;
```

Comment:
This test should be minimized further:

```
trait Foo {}
impl<T> Foo for T {}

fn main() {
    let array = [(); { loop {} }];

    let tup = (7,);
    let x: &dyn Foo = &tup.0;
}
```

Everything else happening in the test is misleading to the actual issue here (tainting infcx via evaluation failure + ambiguous integer infer var + a field access + coercion that doesn't constrain it).